### DISCUSSION TITLE
Enhancing the library with other features

### DISCUSSION THREAD

**Discussion by tarikjabiri** [ognjen-petrovic/js-dxf#39] [Ideas]
> As we create DXF files we wants to parse them and extract information from them to use it after :+1: 

something like :
```javascript
const dxf = DXFManager('file path or dxf plain txt');
const layers = dxf.getLayers();
const entities = dxf.getEntities();
const lines = dxf.getLines();
// so on ...
```
Or we can add to it other entities or layers ...

```javascript
dxf.addLayer('name', 'CENTER', 'Color', 'weight');
dxf.addLine(...);
dxf.addPolyline(...);
// Also change the version of the dxf
dxf.setVersion('AC1027')
```

- And as @kriffe say is better to use Typescript for development 

And I want to contribute when i have time :100: 
because this will help me on a project I want to read dxf files an export them :1st_place_medal:

**Comment by kriffe** [comment]
I think that js-dxf could be used in combination with other very competent dxf parsers to accomplish this. My experience is that https://github.com/bjnortier/dxf is a very good library for reading dxf files.

Not sure that I would put them both in the same. Rather use them together. Just want to make sure that we dont reinvent the wheel.

**Reply by tarikjabiri** [reply]
Hi,
I agree with you
And what I am aiming for is I want to get an Entity object like LineEntity with its Layer its coordinate its color
that will be easy to manipulate.
I am not sure if we can use the library that you motioned and wrap it with our logic so we don't reinvent everything.
if OK I will write a prototype based on the tow lib and see if it will work.
first I need to explore the dxf lib because I never used it.

**Comment by ognjen-petrovic** [comment]
>     * And as @kriffe say is better to use Typescript for development
> 
> 
> And I want to contribute when i have time 💯
> because this will help me on a project I want to read dxf files an export them 🥇

Yes, I agree about the Typescript. I would also like to get rid of the "require" and use vanilla "import" statement.

**Reply by novotechsoft** [reply]
Hi everyone, I wonder if there is any way to set the scaling for the DXF file. In most drafting jobs, we need to set the scale, like 1:100 or 1:1000 (depending on the Imperial or Metric units used). Does this library have this feature?

**Reply by ognjen-petrovic** [reply]
Hi,

No, there is no such feature.
